File: 브루트포스/leetcode1779(Easy)-Find Nearest Point That Has the Same X or Y Coordinate.js
/**
 * @param {number} x
 * @param {number} y
 * @param {number[][]} points
 * @return {number}
 */
var nearestValidPoint = function (x, y, points) {
  const valid = (px, py) => {
    return px === x || py === y
  }

  const mht = (px, py) => {
    return Math.abs(px - x) + Math.abs(py - y)
  }

  let minDist = Infinity
  const validPoints = []
  for (let index = 0; index < points.length; index += 1) {
    const [px, py] = points[index]
    if (!valid(px, py)) {
      continue
    }
    const dist = mht(px, py)
    validPoints.push([index, dist])
    minDist = Math.min(minDist, dist)
  }

  for (const [index, dist] of validPoints) {
    if (minDist === dist) {
      return index
    }
  }
  return -1
}

File: 브루트포스/leetcode1779(Easy)-Find Nearest Point That Has the Same X or Y Coordinate.py
# 간단한 O(N) 순회문제
class Solution:
    def nearestValidPoint(self, x: int, y: int, points: List[List[int]]) -> int:
        def valid(px, py):
            return x == px or y == py

        def mht(px, py):
            return abs(x - px) + abs(y - py)

        min_dist = int(1e9)
        valid_points = []
        for index, coor in enumerate(points):
            px, py = coor
            if not valid(px, py):
                continue
            dist = mht(px, py)
            valid_points.append([index, dist])
            min_dist = min(min_dist, dist)

        for index, dist in valid_points:
            if dist == min_dist:
                return index
        
        return -1